src/responseType.ts

```diff
@@ -3,9 +3,17 @@ type BaseError = {
   message: string;
 };
 
-type FieldError<RequestData> = BaseError & { property: keyof RequestData };
+type Path<ObjectType extends object> = {
+  [Key in keyof ObjectType & (string | number)]: ObjectType[Key] extends object
+    ? `${Key}` | `${Key}.${Path<ObjectType[Key]>}`
+    : `${Key}`;
+}[keyof ObjectType & (string | number)];
```

Comment:
I was wondering, if you do: 
```typescript
type Foo = Path<{
  email: string;
  testDrive?: { date?: string };
}>;
// -> "testDrive" | "testDrive.date" | "email"
```
do you want the testDrive or only the testDrive.date one?

Reply:
I'm interested in both. The reason is - we can have a specific error on nested object fields or a more generic one on the entire nested object. I'd like to support both cases

Reply:
ah got it, nicely solved 👍